airo-typing/airo_typing/__init__.py

```diff
@@ -69,11 +69,67 @@
 shorthand notation is ^C T^B_A, where C is the frame the velocity is measured in, B is the frame the velocity is expressed in.
 """
 
-# manipulator types
+#####################
+# Manipulator types #
+#####################
 
 JointConfigurationType = np.ndarray
 """an (N,) numpy array that represents the joint angles for a robot"""
 
+#########################
+# Single-arm path types #
+#########################
+JointPathType = List[JointConfigurationType]
+""" a list of joint configurations that describe a path in joint space"""
+
+JointPathArrayType = np.ndarray
+""" a (N, DoFs) numpy array that represents a path in joint space"""
+
+PosePathType = List[HomogeneousMatrixType]
+""" a list of homogeneous matrices that describe a path in cartesian space"""
+
+PosePathArrayType = np.ndarray
+""" a (N, 4, 4) numpy array that represents a path in cartesian space"""
+
+TimePathType = List[float]
+""" a list of times that describe a path in time, must be monotonically increasing"""
+
+TimePathArrayType = np.ndarray
+""" a (N,) numpy array that represents a path in time, must be monotonically increasing"""
+
+#######################
+# Dual-arm path types #
+#######################
+DualJointConfigurationType = np.ndarray
```

Comment:
Do we need separate Types for dual arm paths? IMO it is user responsability to know the DOFS of the system he/she is controlling. I think this would scale badly to motion planning with grippers (that can have various DOFs?).

Or is this additional code complexity worth the increased usability?